Spark_practice/spark_tasks.py

```diff
@@ -0,0 +1,299 @@
+            [t.StructField('titleId', t.StringType(), True),
+             t.StructField('ordering', t.IntegerType(), True),
+             t.StructField('title', t.StringType(), True),
+             t.StructField('region', t.StringType(), True),
+             t.StructField('language', t.StringType(), True),
+             t.StructField('types', t.StringType(), True),
+             t.StructField('attributes', t.StringType(), True),
+             t.StructField('isOriginalTitle', t.IntegerType(), True)])
+
+        self.title_crew_schema = t.StructType(
+            [t.StructField('tconst', t.StringType(), True),
+             t.StructField('directors', t.StringType(), True),
+             t.StructField('writers', t.StringType(), True)])
+
+        self.title_basics_schema = t.StructType(
+            [t.StructField('tconst', t.StringType(), True),
+             t.StructField('titleType', t.StringType(), True),
+             t.StructField('primaryTitle', t.StringType(), True),
+             t.StructField('originalTitle', t.StringType(), True),
+             t.StructField('isAdult', t.IntegerType(), True),
+             t.StructField('startYear', t.IntegerType(), True),
+             t.StructField('endYear', t.IntegerType(), True),
+             t.StructField('runtimeMinutes', t.IntegerType(), True),
+             t.StructField('genres', t.StringType(), True)])
+
+        self.title_episode_schema = t.StructType(
+            [t.StructField('tconst', t.StringType(), True),
+             t.StructField('parentTconst', t.StringType(), True),
+             t.StructField('seasonNumber', t.IntegerType(), True),
+             t.StructField('episodeNumber', t.IntegerType(), True)
+             ])
+
+        self.title_principals_schema = t.StructType(
+            [t.StructField('tconst', t.StringType(), True),
+             t.StructField('ordering', t.StringType(), True),
+             t.StructField('nconst', t.StringType(), True),
+             t.StructField('category', t.StringType(), True),
+             t.StructField('job', t.StringType(), True),
+             t.StructField('characters', t.StringType(), True)
+             ])
+
+        self.title_ratings_schema = t.StructType(
+            [t.StructField('tconst', t.StringType(), True),
+             t.StructField('averageRating', t.FloatType(), True),
+             t.StructField('numVotes', t.IntegerType(), True)
+             ])
+
+
+class FilmDB:
+    """Database of films"""
+
+    def __init__(self, schema):
+        self.top_5_of_director = None
+        self.top_actors = None
+        self.top_10_each_genre_decades = None
+        self.top_10_each_genre = None
+        self.sixtys_top_films = None
+        self.last_decade_top_films = None
+        self.all_time_top_films = None
+
+        self.path_to_data = './spark_task_data/'
+        self.spark = SparkSession.builder.appName('FilmDB').getOrCreate()
+
+        self.name_basics = self.spark.read.csv(
```

Comment:
Read logic is the same across all DFs, should it be a separate function without code duplication?

Reply:
Packed 
![image](https://user-images.githubusercontent.com/99332934/174148121-318ddb6c-5376-41fc-8581-4d967e45629f.png)